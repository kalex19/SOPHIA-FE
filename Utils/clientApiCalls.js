export const postList = async newList => {
	const url = `https://evening-dusk-50121.herokuapp.com/api/v1/lists`;
	const options = {
		method: 'POST',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(newList)
	};
	const response = await fetch(url, options);
	if (!response.ok) {
		throw new Error('Could not add new list');
	}
	const list = await response.json();
	return list;
};

export const deleteList = async list_id => {
	const url = `https://evening-dusk-50121.herokuapp.com/api/v1/lists/${list_id}`;
	const options = {
		method: 'DELETE',
		headers: { 'Content-Type': 'application/json' }
	};
	const response = await fetch(url, options);
	if (!response.ok) {
		throw new Error('Could not delete list');
	}
};

export const patchList = async (updatedList, list_id) => {
	const options = {
		method: 'PATCH',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(updatedList)
	};
	const response = await fetch(`https://evening-dusk-50121.herokuapp.com/api/v1/lists/${list_id}`, options);

	if (!response.ok) {
		throw new Error('Could not edit the name of the list');
	}
	const list = await response.json();
	console.log('LIST RESPONSE', list);
	return list;
};

export const postClientTask = async (newTask, list_id, client_id) => {
	const options = {
		method: 'POST',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(newTask)
	};
	const response = await fetch(`https://evening-dusk-50121.herokuapp.com/api/v1/lists/${list_id}/tasks`, options);
	if (!response.ok) {
		throw new Error('Could not add new task');
	}
	const task = await response.json();
	return task;
};

export const deleteClientTask = async (list_id, task_id) => {
	const options = {
		method: 'DELETE',
		headers: { 'Content-Type': 'application/json' }
	};
	const response = await fetch(
		`https://evening-dusk-50121.herokuapp.com/api/v1/lists/${list_id}/tasks/${task_id}`,
		options
	);
	if (!response.ok) {
		throw new Error('Could not delete task');
	}
};

export const fetchCaretakers = async () => {
	const response = await fetch(`https://evening-dusk-50121.herokuapp.com/api/v1/caretakers`);
	if (!response.ok) {
		throw new Error('Could not fetch caretakers');
	} else {
		const caretakers = response.json();
		return caretakers;
	}
};

export const fetchClients = async () => {
	const response = await fetch(`https://evening-dusk-50121.herokuapp.com/api/v1/clients`);
	if (!response.ok) {
		throw new Error('Could not fetch clients');
	} else {
		const clients = response.json();
		return clients;
	}
};

export const patchClientProfile = async (updatedProfile, user_id) => {
	// console.log('updated profile', updatedProfile);
	// console.log('user id', user_id);
	const options = {
		method: 'PATCH',
		headers: { 'Content-Type': 'application/json' },
		body: JSON.stringify(updatedProfile)
	};
	const response = await fetch(`https://evening-dusk-50121.herokuapp.com/api/v1/clients/${user_id}`, options);
	// console.log('response', response)

	if (!response.ok) {
		throw new Error('Could not edit the client profile');
	}
	const profile = await response.json();
	return profile;
};
