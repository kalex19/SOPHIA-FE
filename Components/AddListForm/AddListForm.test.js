import React from 'react';
import { AddListForm } from './AddListForm';
import { shallow } from 'enzyme';
import { mapStateToProps, mapDispatchToProps } from './AddListForm';
import { addList } from '../../actions';
import { postList } from '../../Utils/clientApiCalls';


jest.mock("react-native-gesture-handler", () => {
  return {};
});
jest.mock("expo-av", () => {
  return {};
});
jest.mock("expo-file-system", () => {
  return {};
});
jest.mock("expo-font", () => {
  return {};
});
jest.mock("expo-permissions", () => {
  return {};
});

global.fetch = jest.fn().mockImplementation(() => {
  return Promise.resolve({
    ok: true,
    json: () => Promise.resolve(mockUser)
  });
});

let mockHandleChange = jest.fn();
let mockHandleEditList = jest.fn();
let mockHandleSubmit = jest.fn();
let mockHandleValueChange = jest.fn();
let mockCreateNewList = jest.fn();
let mockPostList = jest.fn();
let wrapper = shallow(<AddListForm 
handleChange = {mockHandleChange}
handleEditList = {mockHandleEditList}
handleSubmit = {mockHandleSubmit}
handleValueChange = {mockHandleValueChange}
createNewList = {mockCreateNewList}
user = {{ role: "client", id: 3}}
/>)

test("AddListForm renders correctly", () => {
  expect(wrapper).toMatchSnapshot();
});

it("should change the state of the list_title when handleChange is invoked", () => {
  wrapper.state.list_title = "";

  let mockName = "new list name";

  wrapper.instance().handleChange(mockName);

  expect(wrapper.state("list_title")).toEqual(mockName);
});

it("should change the state of list_edit_input when handleEditList is invoked", () => {
  wrapper.state.list_edit_input = "";

  let mockName = "new edit input";

  wrapper.instance().handleEditList(mockName);

  expect(wrapper.state("list_edit_input")).toEqual(mockName);
});

it('should call postList when handleSubmit is invoked', () => {
  wrapper.instance().handleSubmit();

  expect(wrapper.postList).toHaveBeenCalled;
})

it('should navigate to set state to list_title: "", caretaker_id: 0, client_id: 0  when handleSubmit is invoked', async () => {
  
  wrapper.state.list_title = "Sample list title"
  wrapper.state.caretaker_id = 1
  wrapper.state.client_id = 1
  await wrapper.instance().handleSubmit();
  expect(wrapper.state('caretaker_id')).toEqual(0);
  expect(wrapper.state('client_id')).toEqual(0);
})

it('should set the state to the user ids for a client or a caretaker', () => {
  let mockUserId = 2;

  wrapper.instance().handleValueChange(mockUserId);

  expect(wrapper.state('caretaker_id')).toEqual(2);
  expect(wrapper.state('client_id')).toEqual(3)
})

describe('mapStateToProps', () => {
  let initialState = {
    lists: [
      {title: 'Hat'},
      {title: 'Hat 2'}
    ],
    user: undefined,
  };

  it('should return a user and an array of lists', () => {
    const mappedProps = mapStateToProps(initialState);
    expect(mappedProps).toEqual(initialState);
  });
});

describe('mapDispatchToProps', () => {
  let initialState = {
    user: "User",
    lists: [
      {title: 'Hat'},
      {title: 'Hat 2'}
    ]
    };
  
    it('calls dispatch with a addList action when addList is called', () => {
      const mockDispatch = jest.fn();
      const newList = { title: 'Hat 3' }
      const mockAction = addList(newList);
  
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.addList(newList);
      expect(mockDispatch).toHaveBeenCalledWith(mockAction);
    });
  });
