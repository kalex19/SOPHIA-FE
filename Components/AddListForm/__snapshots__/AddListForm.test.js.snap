// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddListForm renders correctly 1`] = `ShallowWrapper {}`;
